app/lib/core/view_models/signout_view_model.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:app/core/services/authentication/auth_service.dart';
+import 'package:app/core/services/navigation/nav_service.dart';
+import 'package:app/core/services/service_locator.dart';
+import 'package:app/core/view_models/base_view_model.dart';
+
+class SignOutViewModel extends BaseViewModel {
+  //Inject auth service
+  final AuthenticationService _authenticationService =
+      locator<AuthenticationService>();
+  final NavigationService _navigationService = locator<NavigationService>();
+
+  Future signOut() async {
+    // TODO: Should handle exception error
+    try {
+      await _authenticationService.signOut();
+      _navigationService.pop();
```

Comment:
The asynchronous code allows the pop before print, but I think we really should pop after the print (make navigation last) because traditionally, we would have exited the stack call at pop which would have left this function hanging. Recall the jumps in ENSE 352.

This is working because the UI is separate from executing code (popping is like calling hide), but every time I see this I think it smells.

Reply:
Will switch this  two in my next commit